pisa/stages/osc/nsi_params.py

```diff
@@ -2,12 +2,34 @@
 # date:   June 28, 2017
 """
 NSIParams: Characterize non-standard neutrino interaction coupling strengths
+
+changed by Elisa Lohfink (ellohfin; elohfink@icecube.wisc.edu)
+to include NSI changes made by Thomas Ehrhardt on his branch:
+original version can be found in thehrh/pisa nsi_reparameterisation branch
 """
 
 from __future__ import division
 
 import numpy as np
 
+from pisa import FTYPE
+
+__all__ = ['NSIParams', 'StdNSIParams', 'VacuumLikeNSIParams']
+
+def _set_magnitude_phase(magn_phase_tuple):
+    try:
+        magnitude, phase = magn_phase_tuple
+    except:
+        raise ValueError(
+            'Pass an iterable with two items (magnitude and phase)!'
+        )
+    if magnitude < 0.0 and phase != 0.0:
+        raise ValueError(
+            'Only accepting negative values with a zero phase (real coupling)!'
+        )
+    return magnitude, phase
+
+
 class NSIParams(object):
```

Comment:
This base class seems to be an outdated version compared to https://github.com/thehrh/pisa/blob/nsi_reparameterisation/pisa/stages/osc/nsi_params.py. Can you explain its goal?

Reply:
I'm afraid I don't quite get or see what you mean

Reply:
oh, sorry, now I got it I think

Reply:
Do you mean the part where I stuck to the PISA master way of doing this instead of the NSI branch way? Or do I misunderstand what you mean?

Reply:
Yes. The functionality in this version of the base class is meant to be superseded by the two derived classes, in particular by `StdNSIParams`. You can also compare the docstrings of `NSIParams` here and in https://github.com/thehrh/pisa/blob/nsi_reparameterisation/pisa/stages/osc/nsi_params.py.

Unless you have any particular usage of this (in my eyes outdated) version of `NSIParams` in mind?